corehq/apps/app_manager/views/translations.py

```diff
@@ -99,13 +103,21 @@ def download_bulk_app_translations(request, domain, app_id):
 @require_can_edit_apps
 @get_file("bulk_upload_file")
 def upload_bulk_app_translations(request, domain, app_id):
+    validate = request.POST.get('validate')
     app = get_app(domain, app_id)
-    msgs = process_bulk_app_translation_upload(app, request.file)
-    app.save()
+    msgs = []
+    workbook = read_uploaded_app_translation_file(request.file, msgs)
+    if workbook:
+        if validate:
```

Comment:
The validation code should also be run when the actual upload happens, either here or inside process_bulk_app_translation_upload.

Reply:
hmm, so the way i see is that one would upload a file and check if its good and then simply translate with that same file if all went well. Checking again would ensure if the app has changed in between those two actions..

Reply:
Yeah, I agree that's the most likely workflow, but if validation is quick, I think doing it again is worthwhile. We double up on validation relatively often - e.g., some validation is done both client-side and server-side - for the sake of making HQ more robust.